sdk/containerregistry/azure-containerregistry/azure/containerregistry/aio/_async_container_registry_client.py

```diff
@@ -71,22 +72,20 @@ async def delete_repository(self, repository_name: str, **kwargs: Dict[str, Any]
                 :dedent: 8
                 :caption: Delete a repository from the `ContainerRegistryClient`
         """
-        result = await self._client.container_registry.delete_repository(repository_name, **kwargs)
-        return DeleteRepositoryResult._from_generated(result)  # pylint: disable=protected-access
+        try:
+            await self._client.container_registry.delete_repository(repository, **kwargs)
+        except ResourceNotFoundError:
+            pass
```

Comment:
Is there a particular error code we can validate here before swallowing? Or does the service just give us an empty 404?

Reply:
This was changed to be caught in the generated code (ie `if response.status_code not in [202, 404]: do_error_stuff`). The body is specific with the following content:
`b'{"errors": [{"code": "NAME_UNKNOWN", "message": "repository name not known to registry", "detail": {"name": "not_real_repo"}}]}'`

I'll start a conversation with the acr team.

Reply:
From the service team:
> Yes, 404 for Delete is not found. For other bad request formats, we return 400

Reply:
Yeah - by error code I mean specifically the "name_unknown" part. This just makes it a more targeted suppression in the case a 404 could be returned under other circumstances.
For example:
```python
try:
    ...
except ResourceNotFoundError as e:
    if e.code == 'NAME_UNKNOWN':
        return
    raise
```